src/mqueryfront/src/QueryField.js

```diff
@@ -20,32 +22,33 @@ class QueryField extends Component {
 
     componentDidMount() {}
 
-    selectTaint(newTaint) {
-        this.setState({
-            selectedTaint: newTaint,
-        });
-    }
-
-    describeTaint() {
-        if (this.state.selectedTaint == null) {
-            return "everywhere";
-        }
-        return this.state.selectedTaint;
-    }
+    handleChange = (selectedOption) => {
+        this.setState({ selectedOption: selectedOption });
+    };
 
     handleYaraChanged(value) {
         this.props.updateYara(value);
     }
 
     handleQuery(event, method, priority) {
+        var taints = [];
+
+        if (this.state.selectedOption != null) {
```

Comment:
you don't need to create this list when the default value for state is `[]` I think?

Even then, you can just `var taints = this.state.selectedOption || []`